File: sleepyrouter/providers/openrouter.py
"""OpenRouter provider adapter."""

from .base import BaseProviderAdapter


class OpenRouterProviderAdapter(BaseProviderAdapter):
    def __init__(self) -> None:
        super().__init__(
            name="OpenRouter",
            source="openrouter",
            api_key_env_var="OPENROUTER_API_KEY",
            api_base="https://openrouter.ai/api/v1",
            model_prefix="openrouter",
            extra_headers={
                "HTTP-Referer": "https://github.com/sleepysoong/sleepyrouter",
                "X-OpenRouter-Title": "sleepyrouter",
            },
        )

File: src/providers/openrouter.ts
import { createOpenAICompatible } from "@ai-sdk/openai-compatible";
import type { Provider } from "./base.js";
import { baseURLFrom } from "./base.js";

export const openRouterProvider: Provider = {
  name: "OpenRouter",
  source: "openrouter",
  messageProtocol: "anthropic",
  chatModel(modelId: string, apiKey: string, customFetch?: typeof fetch) {
    const provider = createOpenAICompatible({
      name: "openrouter",
      baseURL: baseURLFrom(
        "SLEEPYROUTER_OPENROUTER_BASE_URL",
        "https://openrouter.ai/api/v1",
      ),
      apiKey,
      headers: {
        "HTTP-Referer": "https://github.com/sleepysoong/sleepyrouter",
        "X-OpenRouter-Title": "sleepyrouter",
      },
      fetch: customFetch,
    });
    return provider.chatModel(modelId);
  },
};
